optuna/trial/_frozen.py

```diff
@@ -546,6 +546,16 @@ def create_trial(
         Please note that this is a low-level API. In general, trials that are passed to objective
         functions are created inside :func:`~optuna.study.Study.optimize`.
 
+    .. note::
+
+        You cannot specify both ``value`` and ``values`` when ``state`` is ``None``
+        or ``TrialState.COMPLETE``.
```

Comment:
Isn't this limitation true independent of the state (or, it should be at least)?

Reply:
yeah, that should be the case.

I added this note because the current latest documentation sounded like both `value` and `values` are required as per the caption below.

![image](https://user-images.githubusercontent.com/16191443/103118928-6788ac80-46b4-11eb-8c73-c4fc60270a47.png)

Reply:
Ah, I see. So, the same phrase and conditioning on the state is currently repeated quite often. I'm wondering if it would be easier to comprehend if put all these conditions under a single note, and simplify the notes, as well as the parameter documentation.

```
.. note::
    When ``state`` is ``TrialState.COMPLETE`` or :obj:`None`, the following parameters must not be :obj:`None`. Otherwise, errors are raised.

  - ``params``
  - ``distributions``
  - ``value`` or ``values``
```

That ``value`` and ``values`` cannot be specified at the same time is further elaborated under `Raises:`.